SSA/Projects/InstCombine/PaperExamples.lean

```diff
@@ -34,12 +34,10 @@ theorem shift_mul:
   rcases A with rfl | A  <;>
   rcases B with rfl | B  <;> (try (simp only [bind, Option.bind, Refinement.refl]; done)) <;>
   by_cases h : w ≤ BitVec.toNat B <;>
-    simp only [ge_iff_le,
-      EffectKind.return_impure_toMonad_eq, Option.pure_def, mul_eq,
-      Option.bind_eq_bind, Option.none_bind, h, ↓reduceIte, Option.none_bind,
-      Option.bind_none, Option.some_bind, Refinement.some_some, Refinement.refl]
-  simp only [Bool.false_eq_true, shiftLeft_eq', false_and, toNat_shiftLeft, toNat_ofNat, _root_.or_self,
-  ↓reduceIte, one_shiftLeft_mul, Refinement.refl]
+    simp only [Bool.false_eq_true, ge_iff_le, toNat_allOnes, false_and, _root_.or_self, ↓reduceIte,
+      shiftLeft_eq', EffectKind.return_impure_toMonad_eq, Option.pure_def, mul_eq,
+      Option.bind_eq_bind, Option.none_bind, Option.bind_none, Option.some_bind, Refinement.refl, h]
+  simp only [one_shiftLeft_mul, Refinement.refl]
```

Comment:
AFAIK when `simp` is final you can just leave `simp` instead of `simp only` (which shall instead be used when non-final)

Reply:
OK, I will change that right now